zeebe/exporters/camunda-exporter/src/test/java/io/camunda/exporter/CamundaExporterIT.java

```diff
@@ -382,4 +334,158 @@ private ExporterResourceProvider mockResourceProvider(
+                        ImmutableDecisionEvaluationRecordValue.builder()
+                            .from(factory.generateObject(DecisionEvaluationRecordValue.class))
+                            .withEvaluatedDecisions(List.of(decisionValue))
+                            .build();
+                    return factory.generateRecord(
+                        ValueType.DECISION_EVALUATION,
+                        r ->
+                            r.withValue((T) decisionRecordValue)
+                                .withTimestamp(System.currentTimeMillis()));
+                  });
+
+      final Function<BpmnElementType, Supplier<Record<T>>> getProcessInstanceRecord =
+          (elementType) ->
+              () ->
+                  recordGenerator(
+                      handler,
+                      () -> {
+                        final ProcessInstanceRecordValue processInstanceRecordValue =
+                            ImmutableProcessInstanceRecordValue.builder()
+                                .from(factory.generateObject(ProcessInstanceRecordValue.class))
+                                .withParentProcessInstanceKey(-1L)
+                                .withBpmnElementType(elementType)
+                                .build();
+                        return factory.generateRecord(
+                            ValueType.PROCESS_INSTANCE,
+                            r ->
+                                r.withIntent(ELEMENT_ACTIVATING)
+                                    .withValue((T) processInstanceRecordValue)
+                                    .withTimestamp(System.currentTimeMillis()));
+                      });
+
+      final Map<Class<?>, Supplier<Record<T>>> customRecordGenerators =
+          Map.of(
+              EventFromJobHandler.class,
+              getJobRecord,
+              ListViewFlowNodeFromJobHandler.class,
+              getJobRecord,
+              ListViewProcessInstanceFromProcessInstanceHandler.class,
+              getProcessInstanceRecord.apply(BpmnElementType.PROCESS),
+              ListViewFlowNodeFromProcessInstanceHandler.class,
+              getProcessInstanceRecord.apply(BpmnElementType.BOUNDARY_EVENT),
+              DecisionEvaluationHandler.class,
+              getDecisionEvalRecord,
+              MetricFromProcessInstanceHandler.class,
+              getProcessInstanceRecord.apply(BpmnElementType.PROCESS),
+              UserTaskProcessInstanceHandler.class,
+              getProcessInstanceRecord.apply(BpmnElementType.PROCESS));
+
+      final Supplier<Record<T>> defaultRecordGen =
+          () ->
+              recordGenerator(
+                  handler,
+                  () ->
+                      factory.generateRecord(
+                          handler.getHandledValueType(),
+                          r -> r.withTimestamp(System.currentTimeMillis())));
+      return () -> customRecordGenerators.getOrDefault(handler.getClass(), defaultRecordGen).get();
```

Comment:
❓ Does this make sense to recreate the suppliers everytime again?

Reply:
No I'll move them into functions.